libs/confman/src/file_storage.cpp

```diff
@@ -0,0 +1,141 @@
+#include <cstdint>
+#include <cstdlib>
+#include <iterator>
+#include <map>
+#include <set>
+#include <system_error>
+
+#include <stduuid/uuid.h>
+
+#include <tfc/utils/pragmas.hpp>
+
+// clang-format off
+// dynamic and clang build would fail if this warning is not deactivated
+PRAGMA_CLANG_WARNING_PUSH_OFF(-Wdate-time)
+#include <pcg_random.hpp>
+#include <pcg_extras.hpp>
+PRAGMA_CLANG_WARNING_POP
+// clang-format on
+
+#include <tfc/confman/file_storage.hpp>
+
+namespace tfc::confman {
+
+/// \brief Retrieves an integer value of an environment variable, if a non-negative value is present on the variable
+/// it is returned. Otherwise a default value is returned.
+/// \param env_variable environment variable to fetch
+/// \param default_value default integer value to be returned if the environment variable is not set or holds an invalid
+/// value. \return value of the environment variable or the default value. \example
+/// get_environment_variable("MAX_CONNECTIONS", 5); If the environment variable "MAX_CONNECTIONS" is set to "10", the
+/// function returns 10. Otherwise, the function returns 5.
+static auto get_environment_variable(std::string const& env_variable, int default_value) -> int {
+  if (const char* value = std::getenv(env_variable.c_str()); value != nullptr) {
+    if (int const value_i = std::stoi(value); value_i >= 0) {
+      return value_i;
+    }
+  }
+  return default_value;
+}
+
+/// \brief Fetches the value of the environment variable "TFC_CONFMAN_MIN_RETENTION_DAYS", default value is 30 days.
+/// \return `std::chrono::days` set in the environment variable
+/// \example get_minimum_retention_days();
+/// If the environment variable "TFC_CONFMAN_MIN_RETENTION_DAYS" is set to "15", the function returns std::chrono::days(15).
+/// Otherwise, the function returns std::chrono::days(30).
+static auto get_minimum_retention_days() -> std::chrono::days {
+  return std::chrono::days(get_environment_variable("TFC_CONFMAN_MIN_RETENTION_DAYS", 30));
+}
+
+/// \brief Fetches the value of the environment variable "TFC_CONFMAN_MIN_RETENTION_COUNT", default value is 4.
+/// \return minimum retention count set in the environment variable
+/// \example get_minimum_retention_count();
+/// If the environment variable "TFC_CONFMAN_MIN_RETENTION_COUNT" is set to "6", the function returns 6.
+/// Otherwise, the function returns 4.
+static auto get_minimum_retention_count() -> int {
+  return get_environment_variable("TFC_CONFMAN_MIN_RETENTION_COUNT", 4);
+}
```

Comment:
I would like to see size_t used as a signature. The value should not 
be negative. What does the user logic do with a negative value?

Reply:
changed this to unsigned int.